src/ring_theory/localization.lean

```diff
@@ -1184,6 +1132,81 @@ end num_denom
 
 end fraction_map
 
+section algebra
+
+section is_integral
+variables {R S} {Rₘ Sₘ : Type*} [comm_ring Rₘ] [comm_ring Sₘ] [algebra R S]
+
+/-- Explicit characterization of the submonoid map in the case of an algebra.
+`S` is made explicit to help with type inference -/
+def algebra_map_submonoid (S) [comm_ring S] [algebra R S]
```

Comment:
Should this be in this file? I think it can go close to other more general facts about algebras. Also, can some of the assumptions be weakened to `semiring`?

Reply:
Yes, it looks like all of this can just be `semiring`/`comm_semiring`, and I've moved it all to `ring_theory/algebra.lean`